hail/python/hailtop/utils/utils.py

```diff
@@ -107,7 +108,7 @@ def digits_needed(i: int) -> int:
     return 1 + digits_needed(i // 10)
 
 
-def grouped(n: int, ls: List[Any]) -> List:
+def grouped(n: int, ls: List[Any]) -> Generator[Any]:
```

Comment:
We want the type to be as specific as we know. What are we generating? i.e. what is the type of `group` in the code below?

Reply:
We are generating a list that could have either integers or strings.

Reply:
Correct, so what should `Any` be in this case then?